tests/test_cli.py

```diff
@@ -418,34 +418,196 @@ def test_update_refresh_private_variables_from_template(
-        ["update", "--project-dir", cookiecutter_dir_input.as_posix()]
-        + ["-c", template_version, '--extra-context={"input":"extra-context"}'],
-        input="v\ny\n",
-    )
+    vtu_cli_input, vtu_cli_expected, vtu_cli_updates = variables_to_update_cli
+    vtu_file_input, vtu_file_expected, vtu_file_updates = variables_to_update_file
+
+    expecting_updates = vtu_cli_updates or vtu_file_updates
+
+    cmd_args = [
+        "update",
+        "--project-dir",
+        cookiecutter_dir_input.as_posix(),
+        "-c",
+        template_version,
+    ]
+
+    if vtu_cli_input:
+        cmd_args.append(f"--variables-to-update={vtu_cli_input}")
+
+    if vtu_file_input:
+        variables_to_update_file = tmp_path / "new-cruft-file.json"
+        with open(variables_to_update_file, "w", encoding="utf-8") as fd:
+            json.dump(vtu_file_input, fd)
+        cmd_args.append(f"--variables-to-update-file={variables_to_update_file}")
+
+    result = cruft_runner(cmd_args, input="v\ny\n")
+
     git_diff_captured = capfd.readouterr()
+
+    print()
+    print("========== STDERR ==========")
+    print(git_diff_captured.err)
+    print("========== STDOUT ==========")
+    print(git_diff_captured.out)
+    print(result.stdout)
+    print("============================")
+
+    expected_input_value = vtu_cli_expected or vtu_file_expected  # CLI takes precedence
+
+    # validate input value used in project
+    if expected_input_value:
+        assert f"+Input from cookiecutter: {expected_input_value}" in git_diff_captured.out
+        assert "-Input from cookiecutter: some-input" in git_diff_captured.out
+    else:
+        assert "-Input from cookiecutter:" not in git_diff_captured.out
+        assert "+Input from cookiecutter:" not in git_diff_captured.out
+
+    # validate template versions
     if template_version == "input-updated":
+        expecting_updates = True
         assert "-Initial" in git_diff_captured.out
         assert "+Updated" in git_diff_captured.out
-    assert "-Input from cookiecutter: some-input" in git_diff_captured.out
-    assert "+Input from cookiecutter: extra-context" in git_diff_captured.out
-    assert "cruft has been updated" in result.stdout
+
+    # validate overall result
+    if expecting_updates:
+        assert "cruft has been updated" in result.stdout
+    else:
+        assert "cruft has been updated" not in result.stdout
+        assert "already up to date" in result.stdout
     assert result.exit_code == 0
 
 
+def test_update_changed_variables_wrong_file(
+    cruft_runner,
+    cookiecutter_dir_input,
+    capfd,
+):
+    result = cruft_runner(
+        [
+            "update",
+            "--project-dir",
+            cookiecutter_dir_input.as_posix(),
+            "-c",
+            "input",
+            f"--variables-to-update-file={cookiecutter_dir_input.as_posix()}/.cruft.json",
+        ],
+        input="v\ny\n",
+    )
+
+    git_diff_captured = capfd.readouterr()
+
+    print()
```

Comment:
Can we remove these print statements?

Reply:
Sure, I can, but bare in mind that:
- this is consistent with existing test case `test_diff_skip_git_dir`
- by default, it's only printed if the test fails
- without this it may be very hard to investigate certain test failures

Please confirm you still want me to remove these.